dll/cpl/appwiz/addons.c

```diff
@@ -420,7 +421,14 @@ static INT_PTR CALLBACK installer_proc(HWND hwnd, UINT msg, WPARAM wParam, LPARA
                 EndDialog(hwnd, 0);
             }
             LeaveCriticalSection(&csLock);
-            return FALSE;
+
+            if (s_hThread)
+            {
+                WaitForSingleObject(s_hThread, 5 * 1000);
+                CloseHandle(s_hThread);
```

Comment:
Not sure I understand the point of this PR. You just close the thread's handle, now a little bit later, but that's it. The underlying thread, if it's still alive and doing stuff, is still running.